hstream/hstream.cabal

```diff
@@ -23,7 +23,24 @@ source-repository head
   type:     git
   location: https://github.com/hstreamdb/hstream
 
+flag ReleaseBuild
+  default:     False
+  description:
+    Enables all optimisations, leads to slower build time and better performance
+
+common shared-properties
+  ghc-options:
+    -Wall -Wcompat -Widentities -Wincomplete-record-updates
+    -Wincomplete-uni-patterns -Wpartial-fields -Wredundant-constraints
+
+  if flag(releasebuild)
+    ghc-options:
+      -feager-blackholing -flate-dmd-anal -flate-specialise
+      -fspec-constr-keen -fstatic-argument-transformation
+      -funbox-strict-fields -O2
```

Comment:
if not enable ``releasebuild``, do we maybe consider adding ``-O0`` to make the compilation faster?

Reply:
we should also enable ``releasebuild`` when building docker images.

Reply:
Yes, you are right. GHC [recommends](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/hints.html#sooner-producing-a-program-more-quickly) compiling without any optimization flag for better time. However, cabal will [add a `-O1` flag by default](https://cabal.readthedocs.io/en/3.4/cabal-project.html#cfg-field-optimization). I will add a flag for better compilation time soon. As for the docker image, it is built during the ci workflow. So I am not sure if it is OK to build a release version every time a PR is merged.